mypy/test/teststubgen.py

```diff
@@ -447,15 +449,15 @@ def test_common_dir_prefix(self) -> None:
         assert common_dir_prefix([]) == '.'
         assert common_dir_prefix(['x.pyi']) == '.'
         assert common_dir_prefix(['./x.pyi']) == '.'
-        assert common_dir_prefix(['foo/bar/x.pyi']) == 'foo/bar'
+        assert common_dir_prefix(['foo/bar/x.pyi']) == str(Path('foo/bar'))
         assert common_dir_prefix(['foo/bar/x.pyi',
-                                  'foo/bar/y.pyi']) == 'foo/bar'
+                                  'foo/bar/y.pyi']) == str(Path('foo/bar'))
         assert common_dir_prefix(['foo/bar/x.pyi', 'foo/y.pyi']) == 'foo'
         assert common_dir_prefix(['foo/x.pyi', 'foo/bar/y.pyi']) == 'foo'
         assert common_dir_prefix(['foo/bar/zar/x.pyi', 'foo/y.pyi']) == 'foo'
         assert common_dir_prefix(['foo/x.pyi', 'foo/bar/zar/y.pyi']) == 'foo'
-        assert common_dir_prefix(['foo/bar/zar/x.pyi', 'foo/bar/y.pyi']) == 'foo/bar'
-        assert common_dir_prefix(['foo/bar/x.pyi', 'foo/bar/zar/y.pyi']) == 'foo/bar'
+        assert common_dir_prefix(['foo/bar/zar/x.pyi', 'foo/bar/y.pyi']) == str(Path('foo/bar'))
+        assert common_dir_prefix(['foo/bar/x.pyi', 'foo/bar/zar/y.pyi']) == str(Path('foo/bar'))
```

Comment:
Update test case to cover the case that triggered the endless loop?

Reply:
The case is already covered, but only for Unix paths. The best solution I came with is to add another test case that does the tests with windows paths and runs on windows only (controlled via `pytest` marks). Not sure whether you agree on this as now both tests `test_common_dir_prefix_unix` and `test_common_dir_prefix_windows` mostly duplicate each other, the only difference being the separator used. I have tried monkeypatching to mimick windows behaviour, but found it to be pretty tedious to do and introduce a lot of boilerplate code that also doesn't add to test code readability.